pkg/component/watch.go

```diff
@@ -129,10 +129,17 @@ func WatchAndPush(client *occlient.Client, componentName string, applicationName
 		return fmt.Errorf("error setting up filesystem watcher: %v", err)
 	}
 	defer watcher.Close()
+	defer close(extChan)
```

Comment:
what's the point of passing in a channel / parameter, couldn't you create this channel within the function instead?

Reply:
The unit testing is using this channel to signal start and exit of this function in order to ensure that the simulations of mock file operations starts after watch has been completely started and the function that detects changes(in UT) triggers graceful exit from watch(non-terminating) once the required changes have been caught